src/useAsyncFn.ts

```diff
@@ -18,17 +18,23 @@ export type AsyncState<T> =
       value: T;
     };
 
-const useAsyncFn = <T>(fn: (...args: any[]) => Promise<T>, deps: DependencyList = []): [AsyncState<T>, () => void] => {
-  const [state, set] = useState<AsyncState<T>>({
-    loading: false,
-  });
+export default function useAsyncFn<
+  Result = any,
+  Args extends any[] = any[],
+  Fn extends Function = (...args: Args | []) => Promise<Result>
+>(
+  fn: Fn,
```

Comment:
Same here, maybe we can just do

```ts
fn: (...args: Args | []) => Promise<Result>
```

and get rid of `Fn`?

Reply:
Alternatively, a fancy way would be to just specify `Fn` and get rid of `Args` and `Result` but instead extract them where needed. Only thing then `Args` are not often needed and users will always have to type `Promise<...>` although it is already implied.

Reply:
can't get rid of Args and Result, or otherwise the calling signature from the returned callback won't be typed or callable, I think. I forgot the specific issue I ran in to

Reply:
Args are needed for an app I'm working on, as I'm doing a lot of parameter passing with useAsyncFn

Reply:
I mean either keep (`Args` + `Restult`) or just `Fn`.

Reply:
ah! turns out that my reason for adding Fn was removed when I return value from the then callback. noice

Reply:
adding more tests now. woosh